File: framework/ecommerce-framework-api/src/main/java/com/sdl/ecommerce/api/model/Breadcrumb.java
package com.sdl.ecommerce.api.model;

/**
 * Breadcrumb
 * For facets its a remove link but for categories is a navigation link
 *
 * @author nic
 */
public interface Breadcrumb {

    /**
     * Breadcrumb title to be used for presentation. Should be localized for current language.
     * @return title
     */
    String getTitle();

    /**
     * If breadcrumb is a category. If the breadcrumb is not a category it represents a facet.
     * @return true if breadcrumb is a category
     */
    boolean isCategory();

    /**
     * Get category reference
     * @return reference
     */
    CategoryRef getCategoryRef();

    /**
     * Get facet
     * @return facet
     */
    Facet getFacet();

}

File: dxa.net/ecommerce-framework-dotnet/SDL.ECommerce.OData/Model/Breadcrumb.cs
using SDL.ECommerce.Api.Model;

namespace SDL.ECommerce.OData
{
    public partial class Breadcrumb : IBreadcrumb
    {
        ICategoryRef IBreadcrumb.CategoryRef
        {
            get
            {
                return this.CategoryRef;
            }
        }

        IFacet IBreadcrumb.Facet
        {
            get
            {
                return this.Facet;
            }
        }

        bool IBreadcrumb.IsCategory
        {
            get
            {
                return (bool) this.IsCategory;
            }
        }
    }
}
